core/src/test/java/org/hibernate/ogm/sessionfactory/SessionFactoryConfig.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.sessionfactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.ogm.cfg.OgmConfiguration;
+import org.hibernate.ogm.utils.TestHelper;
+
+/**
+ * Builder class to create a test configuration
+ * to use within {@link SessionFactoryInlineTest} inline helper.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class SessionFactoryConfig {
```

Comment:
I think I would have called it `SessionFactoryBuilder`.

Reply:
Well, with my proposal, it would be a `SessionFactoryBuilder` because you would build the `SessionFactory` with the `build()` method. Currently, you `build()` method builds 2 objects that are then accessible via getters, which could have been called earlier but you would have null results. I don't think it's a good pattern.

`SessionFactoryTestBuilder` would be an acceptable name if we go with my pattern. I agree if you keep yours, your name is OK.

Try what I suggest with building the `SessionFactory` in the `build()` method. I think you'll see it simplifies things and make everything cleaner. (Unless I'm missing something which is possible too :))

Reply:
You're right